test/ServerRequestFactoryTest.php

```diff
@@ -786,4 +786,40 @@ public function testHonorsHostHeaderOverServerNameWhenMarshalingUrl(): void
         $uri = $request->getUri();
         $this->assertSame('example.com', $uri->getHost());
     }
+
+    /**
+     * @psalm-return iterable<string, array{
+     *     0: string
+     * }>
+     */
+    public function invalidHostHeaders(): iterable
+    {
+        return [
+            'comma' => ['example.com,example.net'],
+            'space' => ['example com'],
+            'tab' => ["example\tcom"],
```

Comment:
Add another case here for entries containing newlines.

Reply:
These are already rejected by `HeaderSecurity` and thus the request completely fails to construct. See: https://github.com/laminas/laminas-diactoros/runs/7151517319?check_suite_focus=true

Reply:
Should the `usage.md` be updated to explain that `ServerRequestFactory` may throw based on user input and any exceptions need to be gracefully handled: https://github.com/laminas/laminas-diactoros/blob/2.12.x/docs/book/v2/usage.md#marshaling-an-incoming-request?

see https://github.com/WoltLab/WCF/issues/4888

Reply:
Not sure about it, but this discussion can be deferred to a different issue